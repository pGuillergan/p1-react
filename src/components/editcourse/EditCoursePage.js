import React, { useState } from "react";
import { toast } from "react-toastify";
import EditCourseForm from './EditCourseForm'
import CoursesPage from './../courses/CoursesPage'


export default function EditCoursePage(props) {
  const [course, setCourse] = useState({
    "course_title": '',
    "courseDescription": '',
    units: '',
  });

  function handleChange(event) {
    setCourse({ ...course, [event.target.name]: event.target.value });
  }

  function handleSubmit(event) {
    event.preventDefault();
    fetch('http://54.215.227.33:3000/updatecourse', {
      method: "PATCH",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(course),
    })
      .then((result) => {
        toast.success("Updated course!");
        props.history.push("/courses");
      })
      .catch((err) => {
        toast.error("Course was not Updated..");
      });
  }

  return (
    <div>
    <h3>Edit existing course</h3>
    <EditCourseForm
        onSubmit={handleSubmit}
        onChange={handleChange}
        course={course}
    />
    <CoursesPage/>
</div>
  
  );
}
