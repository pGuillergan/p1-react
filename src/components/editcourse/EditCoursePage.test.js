import React, { Component } from "react";
import { shallow } from "enzyme";

import EditCoursePage from "./EditCoursePage";

describe("EditPage", () => {
  it("should edit course", () => {
    const editPage = shallow(<EditCoursePage/>);
    expect(editPage).toMatchSnapshot();

    expect(editPage.find('h3').length).toEqual(1);
    expect(editPage.find('EditCourseForm').length).toEqual(1);

  });

});