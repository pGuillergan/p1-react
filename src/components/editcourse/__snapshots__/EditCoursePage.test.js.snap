// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPage should edit course 1`] = `
<div>
  <div
    className="jumbotron"
  >
    <h3>
      Edit existing course
    </h3>
    <EditCourseForm
      course={
        Object {
          "courseDescription": "",
          "course_title": "",
          "units": "",
        }
      }
      onChange={[Function]}
      onSubmit={[Function]}
    />
  </div>
  <hr />
  <CoursesPage />
</div>
`;
